src/shell_manager.cc

```diff
@@ -146,6 +146,8 @@ Vector<String> generate_env(StringView cmdline, const Context& context, GetValue
     static const Regex re(R"(\bkak_(quoted_)?(\w+)\b)");
 
     Vector<String> env;
+    for (const auto& [key, value] : context.options()["env"].get<HashMap<String, String, MemoryDomain::Options>>())
+        env.push_back(format("{}={}", key, value));
```

Comment:
This is an ergonomic abstraction; notably it's incomplete in that it does not allow to unset environment variables, and neither does it allow erase the entire environment (though I don't think there is a use case for that).
A more flexible (but probably not better) approach would be to allow overriding `KAKOUNE_POSIX_SHELL` dynamically, then we can set it to `env ... "$@"`..